engine/src/test/scala/cromwell/engine/workflow/workflowstore/SqlWorkflowStoreSpec.scala

```diff
@@ -151,17 +192,47 @@ class SqlWorkflowStoreSpec extends AnyFlatSpec with CromwellTimeoutSpec with Mat
       } yield ()).futureValue
     }
 
+    it should "find a grouped workflow normally when not excluding" taggedAs DbmsTest in {
+      (for {
+        submissionResponses <- workflowStore.add(excludedGroupSourceFilesCollection)
+        startableWorkflows <- workflowStore.fetchStartableWorkflows(10, "A06", 1.second, excludedGroups = Set.empty)
+        _ = startableWorkflows.map(_.id).intersect(submissionResponses.map(_.id).toList).size should be(1)
+        _ <- workflowStore.deleteFromStore(startableWorkflows.head.id) // Tidy up
+      } yield ()).futureValue
+    }
+
+    it should "honor the excludedGroups parameter for a target group" taggedAs DbmsTest in {
+      (for {
+        submissionResponses <- workflowStore.add(excludedGroupSourceFilesCollection)
+        startableWorkflows <- workflowStore.fetchStartableWorkflows(10, "A07", 1.second, excludedGroups = Set("Zardoz"))
+        _ = startableWorkflows.map(_.id).intersect(submissionResponses.map(_.id).toList) should be(empty)
+        _ <- workflowStore.deleteFromStore(submissionResponses.head.id) // Tidy up
+      } yield ()).futureValue
+    }
+
+    it should "select appropriately with the excludedGroups parameter" taggedAs DbmsTest in {
+      (for {
+        submissionResponsesExcluded <- workflowStore.add(excludedGroupSourceFilesCollection)
+        submissionResponsesIncluded <- workflowStore.add(includedGroupSourceFilesCollection)
+        startableWorkflows <- workflowStore.fetchStartableWorkflows(10, "A08", 1.second, excludedGroups = Set("Zardoz"))
```

Comment:
For completeness (not that I doubt this), it would be nice to check that "fetch 10 excluding X" does the right thing (ie fetches 10 workflows that don't match X, rather than fetching 10 workflows and THEN filtering out the Xs)